src/OrleansGetEventStore/GetEventStoreProvider.cs

```diff
@@ -0,0 +1,304 @@
+
+        private string serviceId;
+
+        private readonly int id;
+        private static int counter;
+
+        public GetEventStoreProvider()
+        {
+            id = Interlocked.Increment(ref counter);
+        }
+
+        public const string HostParameterName = "Host";
+        public const string UsernameParameterName = "Username";
+        public const string PasswordParameterName = "Password";
+        public const string PageSizeParameterName = "PageSize";
+
+        private int pageSize;
+
+        public string DefaultStreamName(Type grainType, GrainReference grainReference)
+        {
+            return string.Format("{0}_{1}", serviceId, grainReference.ToKeyString());
+        }
+
+        public async Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            serviceId = providerRuntime.ServiceId.ToString();
+
+            this.Name = name;
+            this.Log = providerRuntime.GetLogger(string.Format("LogViews.{0}.{1}", GetType().Name, id));
+
+            pageSize = config.GetIntProperty(PageSizeParameterName, 500);
+
+            var connectionString = config.GetProperty(HostParameterName, "localhost:1113");
+            var connectionStringParts = connectionString.Split(':');
+            var hostName = connectionStringParts[0];
+            var hostPort = int.Parse(connectionStringParts[1]);
+            var hostAddress = Dns.GetHostAddresses(hostName).First(a => a.AddressFamily == AddressFamily.InterNetwork);
+            
+            Log.Info($"Init Name={Name} ConnectionString={connectionString} Severity={Log.SeverityLevel}");
+
+            var username = config.GetProperty(UsernameParameterName, "admin");
+            var password = config.GetProperty(PasswordParameterName, "changeit");
+
+            var settings = ConnectionSettings.Create()
+                .KeepReconnecting().KeepRetrying()
+                .SetDefaultUserCredentials(new UserCredentials(username, password));
+
+            this.Connection = EventStoreConnection.Create(settings, new IPEndPoint(hostAddress, hostPort));
+            await this.Connection.ConnectAsync();
+
+            Log.Verbose("Successfully Connected");
+        }
+
+        public Task Close()
+        {
+            this.Connection.Close();
+            return TaskDone.Done;
+        }
+ 
+        private static readonly Dictionary<string,object> emptyCommitHeaders = new Dictionary<string, object>();
+
+        public async Task<bool> Append(string streamName, IEnumerable<KeyValuePair<Guid,object>> events, int? expectedVersion)
+        {
+            var eventsToSave = events.Select(e => ToEventData(e.Key, e.Value, emptyCommitHeaders)).ToArray();
+
+            if (eventsToSave.Length == 0)
+                return true;
+
+            var version = expectedVersion.HasValue
+                ? expectedVersion.Value == 0
+                    ? ExpectedVersion.NoStream
+                    : expectedVersion.Value - 1
```

Comment:
expectedVersion really means expected position, i.e. expected length of the stream prior to appending the event
(our interface is a bit more abstract than the store)